base/array.jl

```diff
@@ -2467,3 +2467,192 @@ end
 
 intersect(itr, itrs...) = _shrink(intersect!, itr, itrs)
 setdiff(  itr, itrs...) = _shrink(setdiff!, itr, itrs)
+
+map_unrolled(call, variables::Tuple{}) = ()
+map_unrolled(call, variables) =
+    call(first(variables)), map_unrolled(call, tail(variables))...
+
+map_unrolled(call, variables1::Tuple{}, variables2::Tuple{}) = ()
+map_unrolled(call, variables1, variables2) =
+    call(first(variables1), first(variables2)),
+    map_unrolled(call, tail(variables1), tail(variables2))...
+
+partial_map(call, fixed, variables::Tuple{}) = ()
+partial_map(call, fixed, variables) =
+    call(fixed, first(variables)), partial_map(call, fixed, tail(variables))...
+
+partial_map(call, fixed, variables1::Tuple{}, variables2::Tuple{}) = ()
+partial_map(call, fixed, variables1, variables2) =
+    call(fixed, first(variables1), first(variables2)),
+    partial_map(call, fixed, tail(variables1), tail(variables2))...
+
+function reduce_unrolled(call, item)
+    item
+end
+function reduce_unrolled(call, item1, item2, rest...)
+    reduce_unrolled(call, call(item1, item2), rest...)
+end
+
+struct Rows{Row, Dimensions, Columns} <: AbstractArray{Row, Dimensions}
+    columns::Columns
+end
+
+function compare_axes(reference_axes, item)
+    axes(item) == reference_axes
+end
+function same_axes(first_column, rest...)
+    reduce_unrolled(&, partial_map(compare_axes, axes(first_column), rest)...)
+end
+function same_axes()
+    true
+end
+
+@propagate_inbounds function Rows{Row, Dimension}(columns::Columns) where {Row, Dimension, Columns}
+    @boundscheck if !same_axes(columns...)
+        throw(DimensionMismatch("All arguments to `Rows` must have the same axes"))
+    end
+    Rows{Row, Dimension, Columns}(columns)
+end
+
+get_model(columns) = first(columns)
```

Comment:
can this have a more telling function name?

Reply:
Sure. What should I call it?

Reply:
I have not yet understood what this function does so this is on you ;-)

Reply:
The dimensions and axes of Rows are defined by its first column. If there are no columns, it defaults to `1:0` (the empty unit range). Does that make sense?

Reply:
Maybe `dummy_row`?

Reply:
Sure